lang/nodejs14/Portfile

```
@@ -111,27 +108,43 @@ pre-configure {
      # Copy zlib headers here because we do not want to prepend the entire
      # ${prefix}/include to the include path (the build will then attempt to use
      # ICU 67 headers)
-     file mkdir ${workpath}/zlib-inc
-     file copy ${prefix}/include/zconf.h ${prefix}/include/zlib.h \
-         ${workpath}/zlib-inc/
+    file mkdir ${workpath}/zlib-inc ${workpath}/cares-inc
+    file copy ${prefix}/include/zconf.h ${prefix}/include/zlib.h \
+        ${workpath}/zlib-inc/
+    system "cp ${prefix}/include/*ares*.h ${workpath}/cares-inc"
+
 }
 
+# use the system libuv instead of the bundled version, for two reasons:
+#   1. originally the macports libuv had fixes for older macs
+#           (like 10.7 panicking when spawning with FD_CLOEXEC)
+#   2. now libuv only supports 10.15 and up
+#         -plus the macports libuv should have all features for the appropriate target.
+#
+configure.args-append   --shared-libuv
 configure.args-append   --without-npm
 configure.args-append   --with-intl=system-icu
-configure.args-append   --shared-openssl
-configure.args-append   --shared-openssl-includes=[openssl::include_dir]
-configure.args-append   --shared-openssl-libpath=[openssl::lib_dir]
 configure.args-append   --shared-zlib
 configure.args-append   --shared-zlib-includes=${workpath}/zlib-inc
 configure.args-append   --shared-zlib-libpath=${prefix}/lib
+configure.args-append   --shared-openssl
+configure.args-append   --shared-openssl-includes=${prefix}/libexec/openssl3/include/openssl
+configure.args-append   --shared-openssl-libpath=${prefix}/libexec/openssl3/lib
```

This port includes the openssl portgroup so it was correct to use its functions to get the include and library path; that should not have been changed.

```suggestion
configure.args-append   --shared-openssl-includes=[openssl::include_dir]
configure.args-append   --shared-openssl-libpath=[openssl::lib_dir]
```

Since it looks like you want to switch from openssl 1.1 to 3, change this line near the top of the Portfile:

```
openssl.branch          1.1
```

---

i actually didn't care what openssl version was used, but you are right i should just use the macros that come with the portgroup. so that's what i did.

i didn't change openssl's inclusion otherwise.